memflow/src/architecture/mmu/translate_data.rs

```diff
@@ -0,0 +1,284 @@
+use super::{ArchMMUSpec, MMUTranslationBase};
+use crate::iter::SplitAtIndex;
+use crate::types::Address;
+use std::cmp::Ordering;
+
+use super::MVec;
+
+pub type TranslateVec<'a> = MVec<'a, TranslationChunk<Address>>;
+pub type TranslateDataVec<'a, T> = MVec<'a, TranslateData<T>>;
+
+unsafe fn shorten_datavec_lifetime<'a: 'b, 'b, O>(
+    r: &'b mut TranslateDataVec<'a, O>,
+) -> &'b mut TranslateDataVec<'b, O> {
+    std::mem::transmute(r)
+}
+
+unsafe fn shorten_pair_lifetime<'a: 't, 'b: 't, 't, O>(
+    r: &'t mut (TranslateVec<'a>, TranslateDataVec<'b, O>),
+) -> &'t mut (TranslateVec<'t>, TranslateDataVec<'t, O>) {
+    std::mem::transmute(r)
+}
+
+#[derive(Debug)]
+pub struct TranslateData<T> {
+    pub addr: Address,
+    pub buf: T,
+}
+
+/*impl<T> TranslateData<T> {
```

Comment:
We should remove commented out code (it can be fetched from git anyways in case we ever need it)